src/main/telemetry/srxl.c

```diff
@@ -160,6 +162,168 @@ bool srxlFrameRpm(sbuf_t *dst, timeUs_t currentTimeUs)
+
+    // HDOP
+    hdop = gpsSol.hdop / 10;
+    if (hdop > 99) hdop = 99;
+    hdopBcd = (dec2bcd_r(hdop));
+
+    // flags
+    if (gpsSol.llh.lat > 0) gpsFlags = gpsFlags | 0x01;   // North
+    if (gpsSol.llh.lon > 0) gpsFlags = gpsFlags | 0x02;   // East
+    if (STATE(GPS_FIX)) gpsFlags = gpsFlags     | 0x28;   // GPS Fix
+    if (gpsSol.llh.altCm < 0) gpsFlags = gpsFlags | 0x80; // Negative altitude
+    if ((gpsSol.llh.lon / GPS_DEGREES_DIVIDER) > 99) 
+        gpsFlags = gpsFlags                     | 0x04;   // Longitude > 99
+    gpsFlags = gpsFlags                         | 0x10;   // Data received bit
+
+    // SRXL frame
+    sbufWriteU8(dst, SRXL_FRAMETYPE_GPS_LOC);
+    sbufWriteU8(dst, SRXL_FRAMETYPE_SID);
+    sbufWriteU16(dst, altitudeLoBcd);
+    sbufWriteU32(dst, latitudeBcd);
+    sbufWriteU32(dst, longitudeBcd);
+    sbufWriteU16(dst, groundCourseBcd);
+    sbufWriteU8(dst, hdopBcd);
+    sbufWriteU8(dst, gpsFlags);
+
+    return true;
+}
+
+/*
+typedef struct 
+{ 
+   UINT8   identifier;                      // Source device = 0x17
+   UINT8   sID;                             // Secondary ID
+   UINT16  speed;                           // BCD, knots, format 3.1
+   UINT32  UTC;                             // BCD, format HH:MM:SS.S, format 6.1
+   UINT8   numSats;                         // BCD, 0-99
+   UINT8   altitudeHigh;                    // BCD, meters, format 2.0 (High bits alt)
+} STRU_TELE_GPS_STAT; 
+*/
+
+#define STRU_TELE_GPS_STAT_EMPTY_FIELDS_COUNT 6
+
+bool srxlFrameGpsStat(sbuf_t *dst, timeUs_t currentTimeUs)
+{
+    uint32_t timeBcd;
+    uint16_t speedKnotsBcd, speedTmp;
+    uint8_t numSatBcd, altitudeHighBcd;
+    dateTime_t dt;
+    bool timeProvided = false;
+
+    UNUSED(currentTimeUs);
+
+    // Number of sats and altitude (high bits)
+    numSatBcd = (gpsSol.numSat > 99) ?  dec2bcd_r(99) : dec2bcd_r(gpsSol.numSat);
+    altitudeHighBcd = dec2bcd_r(gpsSol.llh.altCm / 100000);
+
+    // Speed
+    speedTmp = gpsSol.groundSpeed * 1944 / 1000;
+    if (speedTmp > 9999) speedTmp = 9999;
+    speedKnotsBcd = dec2bcd_r(speedTmp);
+
+    // RTC
+    #ifdef USE_RTC_TIME
+    if (rtcHasTime()) {
+        rtcGetDateTime(&dt);
+        timeBcd = dec2bcd_r(dt.hours);
+        timeBcd = timeBcd << 8;
+        timeBcd = timeBcd | dec2bcd_r(dt.minutes);
+        timeBcd = timeBcd << 8;
+        timeBcd = timeBcd | dec2bcd_r(dt.seconds);
+        timeBcd = timeBcd << 4;
+        timeBcd = timeBcd | dec2bcd_r(dt.millis / 100);
+        timeProvided = true;
+    }
+    #endif
+    timeBcd = (timeProvided) ? timeBcd : 0xFFFFFFFF;
+
+    // SRXL frame
+    sbufWriteU8(dst, SRXL_FRAMETYPE_GPS_STAT);
+    sbufWriteU8(dst, SRXL_FRAMETYPE_SID);
+    sbufWriteU16(dst, speedKnotsBcd);
+    sbufWriteU32(dst, timeBcd);
+    sbufWriteU8(dst, numSatBcd);
+    sbufWriteU8(dst, altitudeHighBcd);
+    sbufFill(dst, 0xFF, STRU_TELE_GPS_STAT_EMPTY_FIELDS_COUNT);
+
+    return true;
```

Comment:
Same as above comment.

Reply:
I have a few ideas about how to solve that. The simplest one is: assuming that (by it's core nature) GPS data changes are quite frequent we could just send GPS frames when GPS fix is acquired and there are at least 6 sats available (Spektrum TX does not display GPS data when there are less than 6 sats). Return false otherwise. Simple, fast and easy to do.

`
if (!STATE(GPS_FIX) || gpsSol.numSat < 6) { return false; }
`

Could that be qualified as an acceptable solution ?

Reply:
Also you need to check if the feature is enabled by the user. It's good that the code is bounded by `#ifdef USE_GPS`, but that only determines whether GPS code is compiled in - not whether the user has actually connected and enabled a GPS receiver. You should also check `featureIsEnabled(FEATURE_GPS)` before committing to sending these telemetry frames. Note that this has nothing to do with whether the GPS is actually configured properly, is working and has a good fix - just that the user has at least enabled the GPS feature.